samples/cpp/intelligent_scissors.cpp

```diff
@@ -34,6 +34,9 @@ static float local_cost(const Point& p, const Point& q, const Mat& gradient_magn
     float fG = gradient_magnitude.at<float>(q.y, q.x);
     float dp;
     float dq;
+    float WEIGHT_LAP_ZERO_CROSS = 0.43;
+    float WEIGHT_GRADIENT_MAGNITUDE = 0.14;
+    float WEIGHT_GRADIENT_DIRECTION = 0.43;
```

Comment:
For constants please use

```cpp
static const float kWeightLapZeroCross = 0.43f;
```
at the beginning if the function

Reply:
> static

Don't use "static" for variables in functions. Compiler may generate bloated code for that with global mutex lock.

> const

Good point.

> camelCase vs capital letters

There is no strict guidance.
Capital letters may help to highlight that this is not an input variable parameter.
Examples:
```
grep -Rn -E 'const.* [A-Z0-9_]+ = -?[0-9\.]+' ./
```